app/scripts/modules/core/src/pipeline/status/ExecutionParameters.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+
+import { IExecution } from 'core/domain';
+
+import './executionStatus.less';
+import './executionParameters.less';
+
+export interface IExecutionParametersProps {
+  execution: IExecution;
+  showingParams: boolean;
+  columnFormatAfter: number;
+}
+
+export interface IExecutionParametersState {
+  parameters: Array<{ key: string; value: any }>;
+}
+
+export class ExecutionParameters extends React.Component<IExecutionParametersProps, IExecutionParametersState> {
+  constructor(props: IExecutionParametersProps) {
+    super(props);
+
+    // these are internal parameters that are not useful to end users
+    const strategyExclusions = ['parentPipelineId', 'strategy', 'parentStageId', 'deploymentDetails', 'cloudProvider'];
+
+    let parameters: Array<{ key: string; value: any }> = [];
+
+    const { execution } = this.props;
+    if (execution.trigger && execution.trigger.parameters) {
```

Comment:
I know you migrated this from existing code, but we should prefer re-evaluating computed values from props in `render` over copying to `state`.  Would you mind refactoring, either in this PR or a separate one?

See https://overreacted.io/writing-resilient-components/#dont-stop-the-data-flow-in-rendering

Reply:
Man `memoize-one` is an awesome library, also great article!

Too bad we don't have react 16.8 yet, otherwise we could have used hooks.